consensus/exit_probability_validator/exit_probability_validator.c

```diff
@@ -6,75 +6,174 @@
   epv->lv = NULL;
+
   return RC_OK;
 }
+
 retcode_t iota_consensus_exit_prob_transaction_validator_is_valid(
-    const exit_prob_transaction_validator_t *ep_validator, trit_array_p tx_hash,
+    exit_prob_transaction_validator_t *epv, trit_array_p tail_hash,
     bool *is_valid) {
   retcode_t ret = RC_OK;
   DECLARE_PACK_SINGLE_TX(tx, tx_models, tx_pack);
 
-  ret = iota_tangle_transaction_load(ep_validator->tangle, TRANSACTION_COL_HASH,
-                                     tx_hash, &tx_pack);
-  if (ret != RC_OK || tx_pack.num_loaded == 0) {
+  ret = iota_tangle_transaction_load(epv->tangle, TRANSACTION_COL_HASH,
+                                     tail_hash, &tx_pack);
+  if (ret != RC_OK) {
+    *is_valid = false;
     return ret;
   }
 
-  // TODO - check
-
-  /*TransactionViewModel transactionViewModel =
-  TransactionViewModel.fromHash(tangle, transactionHash); if
-  (transactionViewModel.getType() == TransactionViewModel.PREFILLED_SLOT) {
-    log.debug("Validation failed: {} is missing in db", transactionHash);
-    return false;
-  } else if (transactionViewModel.getCurrentIndex() != 0) {
-    log.debug("Validation failed: {} not a tail", transactionHash);
-    return false;
-  } else if (!transactionViewModel.isSolid()) {
-    log.debug("Validation failed: {} is not solid", transactionHash);
-    return false;
-  } else if (belowMaxDepth(transactionViewModel.getHash(),
-  milestone.latestSolidSubtangleMilestoneIndex - maxDepth)) {
-    log.debug("Validation failed: {} is below max depth", transactionHash);
-    return false;
-  } else if (!ledgerValidator.updateDiff(myApprovedHashes, myDiff,
-  transactionViewModel.getHash())) { log.debug("Validation failed: {} is not
-  consistent", transactionHash); return false;
+  if (tx_pack.num_loaded == 0) {
+    *is_valid = false;
+    log_error(WALKER_VALIDATOR_LOGGER_ID,
+              "Validation failed, transaction is missing in db\n");
+    return RC_OK;
   }
-  return true;*/
+
   if (tx.current_index != 0) {
     log_error(WALKER_VALIDATOR_LOGGER_ID,
               "Validation failed, transaction is not a tail\n");
+    *is_valid = false;
+    return RC_OK;
+  }
+
+  ret = iota_consensus_ledger_validator_update_diff(
+      epv->lv, &epv->analyzed_hashes, &epv->diff, tail_hash->trits, is_valid);
+  if (ret != RC_OK) {
+    *is_valid = false;
+    return ret;
+  }
+  if (!*is_valid) {
+    log_error(WALKER_VALIDATOR_LOGGER_ID,
+              "Validation failed, tail is inconsistent\n");
+    return RC_OK;
+  }
+
+  bool below_max_depth = false;
+
+  // TODO - lowest_allowed_depth
+  ret = iota_consensus_exit_prob_transaction_validator_below_max_depth(
+      epv, tail_hash, epv->max_depth, &below_max_depth);
+  if (ret != RC_OK) {
+    return ret;
+  }
+
+  if (below_max_depth) {
+    *is_valid = false;
+    log_error(WALKER_VALIDATOR_LOGGER_ID,
+              "Validation failed, tail is below max depth\n");
     return RC_OK;
   }
 
+  // TODO - change schema and update solidity
```

Comment:
can we remove this and move to an issue?

Reply:
done